Gems/Atom/Feature/Common/Code/Source/SplashScreen/SplashScreenPass.cpp

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Atom/RPI.Public/Image/StreamingImage.h>
+#include <Atom/RPI.Public/RPIUtils.h>
+#include <Atom/RPI.Reflect/Asset/AssetUtils.h>
+#include <AzCore/Settings/SettingsRegistryImpl.h>
+#include <AzCore/Time/TimeSystem.h>
+#include <SplashScreen/SplashScreenPass.h>
+
+namespace AZ::Render
+{
+    AZ::RPI::Ptr<SplashScreenPass> SplashScreenPass::Create(const AZ::RPI::PassDescriptor& descriptor)
+    {
+        AZ::RPI::Ptr<SplashScreenPass> pass = aznew SplashScreenPass(descriptor);
+        return pass;
+    }
+
+    SplashScreenPass::SplashScreenPass(const AZ::RPI::PassDescriptor& descriptor)
+        : AZ::RPI::FullscreenTrianglePass(descriptor)
+    {
+    }
+
+    SplashScreenPass::~SplashScreenPass()
+    {
+        Clear();
+    }
+
+    void SplashScreenPass::SetupFrameGraphDependencies(AZ::RHI::FrameGraphInterface frameGraph)
+    {
+        AZ::RPI::RenderPass::SetupFrameGraphDependencies(frameGraph);
+    }
+
+    void SplashScreenPass::InitializeInternal()
+    {
+        AZ::RPI::FullscreenTrianglePass::InitializeInternal();
+
+        auto settingsRegistry = AZ::SettingsRegistry::Get();
+        static const AZStd::string setregPath = "/O3DE/Atom/Feature/SplashScreen";
+
+        if (!(settingsRegistry && settingsRegistry->GetObject(&m_settings, azrtti_typeid(m_settings), setregPath.c_str())))
+        {
+            return;
+        }
+
+        m_splashScreenImage = AZ::RPI::LoadStreamingTexture(m_settings.m_imagePath);
+        m_screenTime = m_settings.m_lastingTime;
+
+        if (!m_splashScreenImage)
+        {
+            return;
+        }
+
+        m_splashScreenImageIndex.Reset();
+        m_splashScreenImageIndex.Reset();
```

Comment:
why reset this twice?
```suggestion
        m_splashScreenImageIndex.Reset();
```

Reply:
My bad. One of them should be the other index.